sdk/core/azure-core/azure/core/pipeline/_base_async.py

```diff
@@ -29,10 +29,17 @@
 from azure.core.pipeline import PipelineRequest, PipelineResponse, PipelineContext
 from azure.core.pipeline.policies import AsyncHTTPPolicy, SansIOHTTPPolicy
 from ._tools_async import await_result as _await_result
-from .transport import AsyncHttpTransport
-
-AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType")
-HTTPRequestType = TypeVar("HTTPRequestType")
+from .transport import (
+    AsyncHttpTransport,
+    AsyncHttpResponse as LegacyAsyncHttpResponse,
+    HttpRequest as LegacyHttpRequest,
+    HttpResponse as LegacyHttpResponse,
+)
+from azure.core.rest import AsyncHttpResponse, HttpRequest, HttpResponse
+
+AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType", LegacyAsyncHttpResponse, AsyncHttpResponse)
+HTTPResponseType = TypeVar("HTTPResponseType", HttpResponse, LegacyHttpResponse)
```

Comment:
Same here, you do not it a concrete type here, the implementation will take care of checking consistency with actual types. The loose TypeVar is better.

Reply:
This TypeVar is also used for policies https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/pipeline/_base.py#L56.

Do you mean we also use loose TypeVar for policies?

Or did I miss something?